### PR TITLE
Migrate remaining `app` scripts from Groovy to Kotlin

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: app/gradle/journeyTest.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+tasks.register("journeyTest") {
+    description = "Runs the journey tests."
+    group = "Verification"
+
+    dependsOn(journeyTestWindowsContainers)
+    dependsOn(journeyTestLinuxContainers)
+}
+
+tasks.named<JourneyTestNamingCheckTask>("checkJourneyTestNaming").configure {
```
Comment (sschuberth) [inline app/gradle/journeyTest.gradle.kts:94]: @charleskorn I'm a bit clueless here as this line triggers

```
Task with name 'checkJourneyTestNaming' not found in project ':app'.
```

However, if I change the line to `tasks.regsiter<JourneyTestNamingCheckTask>("checkJourneyTestNaming") {` I get

```
A problem occurred configuring project ':app'.
> Cannot add task 'checkJourneyTestNaming' as a task with that name already exists.
```

So Gradle seems to be confused about whether the tasks already exists or not... 😄

Any idea how to solve this?

Comment (sschuberth) [reply app/gradle/journeyTest.gradle.kts]: Hmm, looks like wrapping this in `afterEvaluate` does the trick...

Comment (sschuberth) [reply app/gradle/journeyTest.gradle.kts]: As @vampire kindly pointed out, the reason for the need to use `afterEvaluate` here is tat the plugin itself gets only registered during `afterEvaluate`:

https://github.com/batect/batect/blob/08b33266bf9e05b9efbcda6b636a015bc6c6cd2f/buildSrc/src/main/kotlin/batect/buildtools/KotlinPlugin.kt#L183-L190

@charleskorn do you see a chance for fixing that first in the plugin? What's the reason for only registering it at `afterEvaluate`?

**Comment by sschuberth** [comment]
I've moved some of the commits out to https://github.com/batect/batect/pull/1337.

File: app/gradle/journeyTest.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply app/gradle/journeyTest.gradle.kts]: I'll leave it to using `afterEvaluate` for now.

Comment (charleskorn) [reply app/gradle/journeyTest.gradle.kts]: I think the use of `afterEvaluate` was to work around a timing problem (the plugin is applied first, then this script, so the source set would never exist when the plugin is applied). I'll merge this as-is and see if I can re-jig the plugin to just react to the registration of the `journeyTest` source set.

**Comment by charleskorn** [comment]
Thanks again @sschuberth!

File: app/gradle/journeyTest.gradle.kts (hunk 1 above)
Comment (charleskorn) [reply app/gradle/journeyTest.gradle.kts]: https://github.com/batect/batect/commit/1e3e823d381e7e7a436f3a8f4aea41a7040465d6 seems to work and removes the need for `afterEvaluate`.

**Pull request merged**